File: Problem_Solving_Algorithms/diagonal_difference.rb
def diagonalDifference(array)
    pricipal_diagonal = 0
    secondary_diagonal = 0

    array.each_with_index do |array, index|
        pricipal_diagonal += array[index]
        secondary_diagonal += array[-index-1]
    end
    
    (pricipal_diagonal - secondary_diagonal).abs
end

diagonalDifference([[1, 2, 3],[4, 5, 6],[10, 8, 9]])

File: Problem_Solving_Algorithms/JavaScript/diagonal_difference.js
function diagonalDifference(array) {
    let pricipal_diagonal = 0
    let secondary_diagonal = 0

    for (let row = 0; row < array.length; row++) {
        for (let col = 0; col < array.length; col++) {
            if (array[row] === array[col]) {
                pricipal_diagonal += array[row][col]
            }
            if (row + col === array.length - 1) {
                secondary_diagonal += array[row][col]
            }
        }
    }
    return Math.abs(pricipal_diagonal - secondary_diagonal)
}
